### PR TITLE
feat(python-cdk) Install codeflash in CDK dev

### PR DESCRIPTION
**Pull request by girarda**
## What
Installs codeflash to the `dev` extra and add codeflash condig.

Codeflash can be used to inspect Python code and provide suggestions to improve the performance.

[This follow up PR adds a GHA to run Codeflash on PRs](https://github.com/airbytehq/airbyte/pull/44427).

## User Impact
CDK developers will install codeflash in their dev environment

## Can this PR be safely reverted and rolled back?
- [X] YES 💚

### CODE REVIEW

File: airbyte-cdk/python/pyproject.toml [hunk 1]
```diff
@@ -114,3 +115,10 @@ optional_poetry_groups = ["dev"]
 poetry_extras = ["file-based", "sphinx-docs", "vector-db-based"]
 poe_tasks = ["check-ci"]
 mount_docker_socket = true
+
+[tool.codeflash]
+# All paths are relative to this pyproject.toml's directory.
+module-root = "airbyte_cdk"
+tests-root = "unit_tests"
+test-framework = "pytest"
+ignore-paths = []
```
Comment (alafanechere) [inline airbyte-cdk/python/pyproject.toml:124]:
```suggestion
ignore-paths = []

```

**Review by alafanechere — APPROVED** [review]
I think it's safe to merge.
The failing connector tests trigered by the CDK change are due to CAT failures which look unrelated to this code change. (e.g. on `source-zendesk-support`) cc. @maxi297

**Comment by maxi297** [comment]
> CAT failures which look unrelated to this code change

Yes. They are failing on master too. I opened a thread on our side to see what caused this

**Pull request merged**